addons/payment/static/src/js/manage_form.js

```diff
@@ -152,6 +118,52 @@ odoo.define('payment.manage_form', require => {
             });
         },
 
+        /**
+         * Build the confirmation dialog based on the linked records information.
+         *
+         * @private
+         * @param {Array} linkedRecordsInfo - The list of information about linked records
+         * @return {object}
+         */
+        _buildConfirmationDialog: function (linkedRecordsInfo, onClick) {
+            const $dialogContentMessage = $(
+                '<span>', {text: _t("Are you sure you want to delete this payment method?")}
+            );
+            if (linkedRecordsInfo.length > 0) { // There are documents linked to the token, list them
+                $dialogContentMessage.append($('<br>'));
+                $dialogContentMessage.append($(
+                    '<span>', {text: _t("It is currently linked to the following documents:")}
+                ));
+                const $documentInfoList = $('<ul>');
+                linkedRecordsInfo.forEach(documentInfo => {
+                    $documentInfoList.append($('<li>').append($(
+                        '<a>', {
+                            href: documentInfo.url,
+                            target: '_blank',
+                            title: documentInfo.description,
+                            text: documentInfo.name
+                        }
+                    )));
+                });
+                $dialogContentMessage.append($documentInfoList);
+            }
+            let diag = new Dialog(this, {
```

Comment:
No needs to create a new variable for that ^^
```suggestion
            return new Dialog(this, {
```